aptos-move/framework/aptos-token-objects/sources/token.move

```diff
@@ -89,18 +100,34 @@ module aptos_token_objects::token {
 
         let collection_addr = collection::create_collection_address(&creator_address, &collection_name);
         let collection = object::address_to_object<Collection>(collection_addr);
-        let id = collection::increment_supply(&collection, signer::address_of(&object_signer));
+
+        let deprecated_index = 0;
+        let index = if (features::concurrent_token_v2_enabled()) {
+            option::destroy_with_default(
```

Comment:
I didn't completely understand what `destroy_with_default` is doing here. Are you incrementing total supply in the collection and destroying it? 
If concurrent_token_v2_enabled feature flag is enabled, then why are we assigning `index = 0` instead of `index = aggregator_v2::snapshot(total_supply)`?